upstairs/src/volume.rs

```diff
@@ -1238,25 +1246,167 @@ impl Volume {
                 "Only a single sub_volume is supported"
             )
         }
-        // Sub volumes must all be VolumeConstructionRequest::Region
+
+        // We support replacement for read_only_parent targets with the
+        // same rules as sub_volume targets, however, for any single VCR
+        // update, only one of the sub_volume or read_only_parent targets
+        // can change.
+        let rop_result = match Self::compare_vcr_region_for_replacement(
+            o_read_only_parent,
+            n_read_only_parent,
+        ) {
+            Err(e) => {
+                warn!(log, "read_only_parent error: {}", e);
+                return Err(e);
+            }
+            Ok(rr) => rr,
+        };
+
+        // Determine the sub_volume difference if there is any.
+        let sub_volume_result = match Self::compare_vcr_region_for_replacement(
+            Some(Box::new(o_sub_volumes[0].clone())),
+            Some(Box::new(n_sub_volumes[0].clone())),
```

Comment:
How about making `compare_vcr_region_for_replace` take an `Option<&VolumeConstructionRequest>`, to avoid these shenanigans?  Here's the complete diff:
```diff
diff --git a/upstairs/src/volume.rs b/upstairs/src/volume.rs
index 782dd6d0c2...841d0d90f7 100644
--- a/upstairs/src/volume.rs
+++ b/upstairs/src/volume.rs
@@ -1252,8 +1252,8 @@
         // update, only one of the sub_volume or read_only_parent targets
         // can change.
         let rop_result = match Self::compare_vcr_region_for_replacement(
-            o_read_only_parent,
-            n_read_only_parent,
+            o_read_only_parent.as_deref(),
+            n_read_only_parent.as_deref(),
         ) {
             Err(e) => {
                 warn!(log, "read_only_parent error: {}", e);
@@ -1264,8 +1264,8 @@

         // Determine the sub_volume difference if there is any.
         let sub_volume_result = match Self::compare_vcr_region_for_replacement(
-            Some(Box::new(o_sub_volumes[0].clone())),
-            Some(Box::new(n_sub_volumes[0].clone())),
+            Some(&o_sub_volumes[0]),
+            Some(&n_sub_volumes[0]),
         ) {
             Err(e) => {
                 warn!(log, "sub_volume has invalid differences: {}", e);
@@ -1364,8 +1364,8 @@
     // Any other difference is an error, and an error returned here means the
     // VCRs are incompatible in a way that prevents one from replacing another.
     fn compare_vcr_region_for_replacement(
-        opt_o_vol: Option<Box<VolumeConstructionRequest>>,
-        opt_n_vol: Option<Box<VolumeConstructionRequest>>,
+        opt_o_vol: Option<&VolumeConstructionRequest>,
+        opt_n_vol: Option<&VolumeConstructionRequest>,
     ) -> Result<VCRDelta, CrucibleError> {
         if opt_n_vol.is_some() {
             if opt_o_vol.is_none() {
```